Source/WebCore/platform/graphics/nicosia/NicosiaBuffer.cpp

```diff
@@ -143,35 +150,66 @@ void UnacceleratedBuffer::waitUntilPaintingComplete()
 }
 
 #if USE(SKIA)
-Ref<Buffer> AcceleratedBuffer::create(sk_sp<SkSurface>&& surface, Flags flags)
+Ref<Buffer> AcceleratedBuffer::create(Ref<BitmapTexture>&& texture, Flags flags)
 {
-    return adoptRef(*new AcceleratedBuffer(WTFMove(surface), flags));
+    return adoptRef(*new AcceleratedBuffer(WTFMove(texture), flags));
 }
 
-AcceleratedBuffer::AcceleratedBuffer(sk_sp<SkSurface>&& surface, Flags flags)
+AcceleratedBuffer::AcceleratedBuffer(Ref<BitmapTexture>&& texture, Flags flags)
     : Buffer(flags)
+    , m_texture(WTFMove(texture))
```

Comment:
I think this constructor could just take the BitmapTexture and get the buffer flags from the BitmapTexture flags. It simplifies the caller and ensure there's no mismatch.

Reply:
Fixed.

Reply:
The create() function no longer takes the flags, but computes them, and passes them to AcceleratedBuffer constructor, which forwards them to Buffer constructor.